app.js

```diff
@@ -0,0 +1,130 @@
+//For secure connection:
+// const fs = require('fs');
+const { Pool } = require("pg");
+
+// Configure the database connection.
+
+const config = {
+  user: "max",
+  password: "roach",
+  host: "localhost",
+  database: "bank",
+  port: 26257,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+  //For secure connection:
+  /*ssl: {
+        ca: fs.readFileSync('/certs/ca.crt')
+            .toString()
+    }*/
+};
+
+// Create a connection pool
+
+const pool = new Pool(config);
+
+// Wrapper for a transaction.  This automatically re-calls the operation with
+// the client as an argument as long as the database server asks for
+// the transaction to be retried.
+
+async function retryTxn(n, max, client, operation, callback) {
+  await client.query("BEGIN;");
+  while (true) {
+    n++;
+    if (n === max) {
+      throw new Error("Max retry count reached.");
+    }
+    try {
+      await operation(client, callback);
+      await client.query("COMMIT;");
+      return;
+    } catch (err) {
+      if (err.code != "40001") {
+        return callback(err);
+      } else {
+        console.log("Transaction failed. Retrying transaction.");
+        console.log(err.message);
+        await client.query("ROLLBACK;", () => {
+          console.log("Rolling back transaction.");
+        });
+        await new Promise((r) => setTimeout(r, 2 ** n * 1000));
+      }
+    }
+  }
+}
+
+// This function is called within the first transaction. It creates a table and inserts some initial values.
+
+async function initTable(client, callback) {
+  await client.query(
+    "CREATE TABLE IF NOT EXISTS accounts (id INT PRIMARY KEY, balance INT);",
+    callback
+  );
+  await client.query(
+    "INSERT INTO accounts (id, balance) VALUES (1, 1000), (2, 250);",
+    callback
+  );
+  await client.query("SELECT id, balance FROM accounts;", callback);
+}
+
+async function transferFunds(client, callback) {
+  const from = 1;
+  const to = 2;
+  const amount = 100;
+  var statement = "SELECT balance FROM accounts WHERE id = $1 ;";
```

Comment:
Instead of reassigning `statement`, I would recommend creating unique variables for each one. For example:
```ts
const selectBalanceStatement = "SELECT balance FROM accounts WHERE id = $1 ;";
...
const updateBalanceStatement = "UPDATE accounts SET balance = balance - $1 WHERE id = $2 ;";
```

Reply:
Done.